crates/store/re_chunk_store/src/query.rs

```diff
@@ -413,3 +560,201 @@ impl ChunkStore {
     }
 }
+
+// Queries returning `usize` and `bool`
+impl ChunkStore {
+    /// Returns the number of events logged for a given component on the given entity path across all timelines.
+    pub fn num_events_on_timeline_for_component(
+        &self,
+        timeline: &Timeline,
+        entity_path: &EntityPath,
+        component_name: ComponentName,
+    ) -> usize {
+        re_tracing::profile_function!();
+
+        self.query_id.fetch_add(1, Ordering::Relaxed);
+
+        self.num_events_for_static_component(entity_path, component_name)
+            + self.num_events_on_timeline_for_temporal_component(
+                timeline,
+                entity_path,
+                component_name,
+            )
+    }
+
+    /// Returns the min and max times at which data was logged for an entity
+    /// on a specific timeline.
+    pub fn time_range_for_entity_on_timeline(
+        &self,
+        timeline: &Timeline,
+        entity_path: &EntityPath,
+    ) -> Option<ResolvedTimeRange> {
+        re_tracing::profile_function!();
+
+        self.query_id.fetch_add(1, Ordering::Relaxed);
+
+        let temporal_chunk_ids_per_timeline =
+            self.temporal_chunk_ids_per_entity.get(entity_path)?;
+        let chunk_id_sets = temporal_chunk_ids_per_timeline.get(timeline)?;
+
+        let start = chunk_id_sets.per_start_time.first_key_value()?.0;
+        let end = chunk_id_sets.per_end_time.last_key_value()?.0;
+
+        Some(ResolvedTimeRange::new(*start, *end))
+    }
+
+    /// Returns the number of events logged for an entity on a specific timeline.
+    pub fn num_events_on_timeline_for_all_components(
+        &self,
+        timeline: &Timeline,
+        entity_path: &EntityPath,
+    ) -> usize {
+        re_tracing::profile_function!();
+
+        self.query_id.fetch_add(1, Ordering::Relaxed);
+
+        let mut total_events = 0;
+
+        if let Some(static_chunks_per_component) = self.static_chunk_ids_per_entity.get(entity_path)
+        {
+            for chunk in static_chunks_per_component
+                .values()
+                .filter_map(|id| self.chunks_per_chunk_id.get(id))
+            {
+                total_events += chunk.num_events_cumulative();
+            }
+        }
+
+        if let Some(chunk_ids) = self
+            .temporal_chunk_ids_per_entity
+            .get(entity_path)
+            .and_then(|temporal_chunks_events_per_timeline| {
+                temporal_chunks_events_per_timeline.get(timeline)
+            })
+        {
+            for chunk in chunk_ids
+                .per_start_time
+                .values()
+                .flat_map(|ids| ids.iter().filter_map(|id| self.chunks_per_chunk_id.get(id)))
+            {
+                total_events += chunk.num_events_cumulative();
+            }
+        }
+
+        total_events
+    }
+
+    /// Returns the number of times a static component was logged to an entity.
+    pub fn num_events_for_static_component(
```

Comment:
There's no such thing as a "static component" -- a component is just a type, it doesn't have any temporality attached to it.

```suggestion
    /// Returns the number of static events for a given component.
    pub fn num_static_events_for_component(